From cceb17a9af73fbfcb670a824f657d5dcb24d0dae Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam@afuera.me.uk>
Date: Sat, 4 May 2019 15:53:07 +0200
Subject: [PATCH] build: libtracker-sparql C code depends on libtracker-sparql
 Vala code

We need to express this dependency, otherwise the C code might fail to
build with this kind of error:

    ../src/libtracker-sparql/tracker-notifier.c:89:10: fatal error: tracker-generated-no-checks.h: No such file or directory

The `tracker-generated-no-checks.h` file is generated by Vala.

Hopefully fixes https://gitlab.gnome.org/GNOME/tracker/issues/96
---
 src/libtracker-sparql/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libtracker-sparql/meson.build b/src/libtracker-sparql/meson.build
index 65985baad..30325a37e 100644
--- a/src/libtracker-sparql/meson.build
+++ b/src/libtracker-sparql/meson.build
@@ -76,6 +76,7 @@ libtracker_sparql_intermediate_c = static_library('tracker-sparql-intermediate-c
     enums_c, enums_h,
     libtracker_sparql_c_sources,
     dependencies: [tracker_common_dep, json_glib],
+    link_with: libtracker_sparql_intermediate_vala,
 )
 
 sparqlinc = [include_directories('.'), meson.current_build_dir()]
-- 
2.18.1

